### PR TITLE
Fix for REGISTRY-3224 diff-view implimentation to store

### PR DESCRIPTION
**Pull request by tmkasun**
These are the changes included in this PR:
- Move greg-diff-api to separate extension
- Add  greg-diff-api in both publisher and store extentions
- Copy greg-diff extention to store 
- Update p2 repo gen , bin.xml to copy greg-diff  and greg-diff-api extensions to relevant directories when building the pack
- Update store greg-publisher-api to get versions

### CODE REVIEW

File: modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ */
+$('.container-fluid').resize(function(){
+    setViewPanelsHeight();
+});
+
+/*
+ * Set view panels height relatively to the window
+ */
+function setViewPanelsHeight(){
+    if($(window).width() > 768){
```
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js]: Formatting issue

File: modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    else{
+        $(viewPanel).css('height', 'auto');
+    }
+    $('.CodeMirror, .CodeMirror-merge').height(($(viewPanel).height()-80));
+
+    /* Fix for code mirror diff view panels width unbalance issue */
+    var CodeMirrorSizerMinWidth = '500px';
+    $('.CodeMirror-sizer').each(function(){
+        if($(this).css('min-width') < CodeMirrorSizerMinWidth){
```
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js]: Formatting issue

File: modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            CodeMirrorSizerMinWidth = $(this).css('min-width');
+        }
+    });
+    $('.CodeMirror-sizer').css('min-width', CodeMirrorSizerMinWidth);
+    $('.CodeMirror-vscrollbar').scrollTop(2);
+
+    /* Fix for code mirror scrollbar not shows on window load */
+    if($('.CodeMirror').height() < $('.CodeMirror-code').height()){
```
Comment (mushthaq33) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js]: Formatting issue

Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/js/diff.js]: Thanks @mushthaq33 , This has been already fixed in [this commit](https://github.com/wso2/product-greg/commit/1046e37150e124ef8e55ccb91e961af3ee79fafe)

File: modules/es-extensions/publisher/extensions/app/greg-diff-api/app.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    };
+};
+
+
+app.apiHandlers = function(ctx) {
+    return {
+        onApiLoad: function() {
+            if ((ctx.isAnonContext) && (ctx.endpoint.secured)) {
+                print('{ error:"Authentication error" }'); //TODO: Fix this to return a proper status code
```
Comment (thushara35) [inline modules/es-extensions/publisher/extensions/app/greg-diff-api/app.js]: Did you able to fix the TODO?

File: modules/es-extensions/publisher/extensions/app/greg-diff-api/modules/governance-diff-api.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thushara35) [inline modules/es-extensions/publisher/extensions/app/greg-diff-api/modules/governance-diff-api.js]: Year??

File: modules/es-extensions/store/extensions/app/greg-diff-api/app.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thushara35) [inline modules/es-extensions/store/extensions/app/greg-diff-api/app.js]: year??

File: modules/es-extensions/store/extensions/app/greg-diff/themes/store/css/diff.css [hunk 7]
```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thushara35) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/css/diff.css]: year??

Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg-diff/themes/store/css/diff.css]: Thanks @thushara35 , I have updated the copyright year in [this commit](https://github.com/wso2/product-greg/pull/854/commits/5f9aea00f943aca4c6b5ce3ca67c1e976cd71ef3)

File: modules/es-extensions/store/extensions/app/greg-diff-api/app.js (hunk 6 above)
Comment (tmkasun) [inline modules/es-extensions/store/extensions/app/greg-diff-api/app.js]: Thanks @thushara35 , I have updated the copyright year in [this commit](https://github.com/wso2/product-greg/pull/854/commits/1ff96980dd4b07ff33d878458c4ef1722c6871df)

File: modules/es-extensions/publisher/extensions/app/greg-diff-api/modules/governance-diff-api.js (hunk 5 above)
Comment (tmkasun) [inline modules/es-extensions/publisher/extensions/app/greg-diff-api/modules/governance-diff-api.js]: Thanks @thushara35 , I have updated the copyright year in [this commit](https://github.com/wso2/product-greg/pull/854/commits/77baf80d16294c4df8bed39d18eb41cd70fa0fd0)

File: modules/es-extensions/publisher/extensions/app/greg-diff-api/app.js (hunk 4 above)
Comment (tmkasun) [inline modules/es-extensions/publisher/extensions/app/greg-diff-api/app.js]: Thanks @thushara35 , I have removed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/c82d1167588d480afbfa6def40616714b1fc3eac)

**Pull request merged**